var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -41,7 +43,8 @@ class Dyninst(CMakePackage):
 
     # Dyninst uses elf@1 (elfutils) starting with 9.3.0, and used
     # elf@0 (libelf) before that.
-    depends_on('elf@1', type='link', when='@9.3.0:')
+    depends_on('elf@0.173:', type='link', when='@10.2.0:')
```

Comment:
I believe `elf@0.173:` includes both `elf@0` and `elf@1`, meaning any version of either elfutils or libelf will work.

Reply:
I reverted to the old, unconstrained one.

Reply:
`elf@0.173:` is awkward and ambiguous.  But `elfutils@0.173:` should
be clear and work.  That's what Adam and I are suggesting.

`elf@0` and `elf@1` are awkward for version numbers, but I don't know
if I'd really call a bug.

Sorry if we're torturing you  .... :-)

Reply:
Using `depends_on('elfutils@0.173:', type='link', when='@10.2.0:')` leads to a concretization error/bug. I'll submit that later. I've reverted to using the old, unconstrained `elf@1` for now.